### ISSUE TITLE
data-disable-with not working in forms with method GET

### ISSUE THREAD

**Issue by coorasse** [hotwired/turbo#363]
As quickly mentioned in https://github.com/hotwired/turbo-rails/issues/135,
when I have a form with method=get and a submit button that makes use of data-disable-with, Rails UJS does not play nice with turbo.

@kirillplatonov suggested in https://github.com/hotwired/turbo-rails/issues/135#issuecomment-814413558 a "polyfill" between rails ujs and turbo (see also https://github.com/hotwired/turbo-rails/pull/93)

Unfortunately none of these events are triggered because the form has method=get, so no `"turbo:submit-start"` and `"turbo:submit-end" events. 😢  

An event `turbo:before-fetch-request` is triggered, but on document level and I don't find a way to find out who originally triggered the event.

Minimal reproduction steps can be found here: https://turbo-showcase.herokuapp.com/home/issues.
(not really minimal but it was the fastest way I had to show it).
Source code: https://github.com/renuo/turbo-showcase/commit/7d64f86361253375767c453fbaa37ac72b8fa992

There I have the following:

```
<%= turbo_frame_tag 'form-frame' do %>
  <h5>Form with method GET</h5>

  <p>
    The data-disable-with does not work
  </p>
  <%= form_for @user, url: '/home/form_get_submission', method: :get do |f| %>
    <%= f.button 'Submit this form', class: 'btn btn-primary', data: { disable_with: spinner } %>
  <% end %>
<% end %>
```

and the "polyfill" in place with:

```
delegate(document, Rails.formSubmitSelector, 'turbo:submit-start', disableElement);
delegate(document, Rails.formSubmitSelector, 'turbo:submit-end', enableElement);
```

**Comment by kirillplatonov** [comment]
I would suggest using Stimulus instead of Rails UJS to disable buttons. I use similar code in my apps:
```javascript
import { Controller } from 'stimulus'

export default class extends Controller {
  static values = { with: String }

  handle(event) {
    const button = event.currentTarget
    const disableWith = this.withValue

    if (button.dataset.disabled) {
      event.preventDefault()
    } else {
      button.dataset.disabled = true
      button.innerHTML = disableWith
    }
  }
}
```

```erb
<%= f.button 'Submit this form',
  class: 'btn btn-primary',
  data: { controller: 'disable', disable_with_value: spinner, action: 'disable#handle' } %>
```

**Comment by coorasse** [comment]
Thanks for the suggestion @kirillplatonov , how do you re-enable the button afterwards?

**Comment by kirillplatonov** [comment]
I don't re-enable it since as a response Turbo will usually either replace the entire form frame to show validation error or will make a redirect.

**Comment by jacobdaddario** [comment]
Hey! Yeah, to my knowledge Turbo is incompatible with those Rails UJS behaviors. I’ve been using a disable Stimulus controller as well, but recently I saw it suggested that disable should be in Turbo.

I’ve opened a PR for such behavior at #384 .

**Comment by jacobdaddario** [comment]
With #386 being merged I believe this can now be closed. It adds disable-with behavior into Turbo by default.

**Comment by brunoprietog** [comment]
Addressed in #869
